prov/gni/include/gnix_buddy_allocator.h

```diff
@@ -74,10 +74,14 @@ static inline uint32_t __gnix_buddy_log2(uint32_t v)
 			      (IS_NOT_POW_TWO(LEN)) ? (((LEN) << 1) & ~(LEN)) :\
 			      (LEN))
 
-/* Find the bitmap index for block X */
-#define BITMAP_INDEX(X, BASE, MIN_LEN, LEN) ((size_t) ((X) - (BASE)) /\
-					     (MIN_LEN) + 2 * __gnix_buddy_log2\
-					     ((LEN) / (MIN_LEN)))
+/* Find the bitmap index for block X of size X_LEN */
+#define BITMAP_INDEX(X,\
+		     X_LEN,\
+		     BASE,\
+		     BASE_LEN,\
+		     MIN_LEN) (((size_t) ((X) - (BASE)) / (size_t) (X_LEN)) + \
+			       (BASE_LEN) / ((MIN_LEN) / 2) - (BASE_LEN) / \
+			       ((X_LEN) / 2))
 
```

Comment:
the linus code style checker was probably complaining here?
If so,  I  would suggest for purposes of this PR, move your .git/hooks/pre-commit to .git/hooks/pre-commit.keep
and then you can put the definition again in 3 lines.  I think this is a case of linus code checker making things messy rather than better.  Don't forget to move the pre-commit.keep back to pre-commit after doing your commits though.